test/validation/api/scheduler/scheduler.c

```diff
@@ -2348,6 +2347,130 @@ static void scheduler_test_order_wait_1_thread(void)
 	CU_ASSERT(odp_queue_destroy(queue) == 0);
 }
 
+static int order_wait_helper(void *arg ODP_UNUSED)
+{
+	odp_event_t ev;
+
+	ev = odp_schedule(NULL, ODP_SCHED_WAIT);
+	if (ev == ODP_EVENT_INVALID) {
+		odp_atomic_store_rel_u32(&globals->order_wait.helper_ready, 1);
+		return 0;
+	}
+	odp_event_free(ev);
+
+	odp_atomic_store_rel_u32(&globals->order_wait.helper_active, 1);
+	odp_atomic_store_rel_u32(&globals->order_wait.helper_ready, 1);
+
+	/* Wait that the main thread can attempt to overtake us */
+	odp_time_wait_ns(ODP_TIME_SEC_IN_NS);
+
+	odp_atomic_store_rel_u32(&globals->order_wait.helper_active, 0);
+
+	/* We are not interested in further events */
+	odp_schedule_pause();
+	/* Release context */
+	while ((ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT))
+	       != ODP_EVENT_INVALID) {
+		/* We got an event that was meant for the main thread */
+		odp_event_free(ev);
+	}
+
+	return 0;
+}
+
+static void scheduler_test_order_wait(void)
```

Comment:
This could be "scheduler_test_order_wait_2_thread".

Reply:
I can change that.